src/pickBy.ts

```diff
@@ -0,0 +1,47 @@
+import { purry } from './purry';
+
+/**
+ * Returns a partial copy of an object containing only the keys that satisfy the supplied predicate.
+ * @param object the target object
+ * @param pred predicate
+ * @signature R.pickBy(object, (value, key) => true )
+ * @example
+ *    R.pickBy({ a: 1, b: 2, c: 3, d: 4 }, (value) => value === 1 || value === 4) // => { a: 1, d: 4 }
+ * @data_first
+ * @category Object
+ */
+export function pickBy<T extends object>(
+  object: T,
+  pred: (value: any, key: string) => boolean
```

Comment:
those types are definitely too wide. Please have a look at how `mapValues` is typed to get inspiration.